### PR TITLE
ui: Convert most integration tests from wiremock to axum

### PR DESCRIPTION
**Pull request by jplatte**
Seems to be about the same amount of boilerplate but we can use channels and other things to delay responses now.

### CODE REVIEW

**Review by Hywan — APPROVED** [review]
Look at this guy having fun with axum ;-).

File: crates/matrix-sdk-ui/tests/integration/axum.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+pub fn test_client_builder(routes: axum::Router) -> ClientBuilder {
+    let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), 0);
+    let server = axum::Server::bind(&addr).serve(routes.into_make_service());
+
+    let client_builder = Client::builder()
+        .homeserver_url(format!("http://{}", server.local_addr()))
+        .server_versions([MatrixVersion::V1_7]);
+
+    spawn(async move {
+        server.await.unwrap();
```
Comment (Hywan) [inline crates/matrix-sdk-ui/tests/integration/axum.rs:46]: If the server never stops, the spawned task will never stop. Is it what we want?

Comment (jplatte) [reply crates/matrix-sdk-ui/tests/integration/axum.rs:46]: It should be stopped when the tokio runtime is stopped at the end of every test.

**Comment by jplatte** [comment]
Doesn't seem like we'll be going ahead with this as-is given internal discussions and it's bitrotting, so might as well be closed.

**Pull request closed without merging**